src/Uno.UI/UI/Xaml/UIElement.wasm.cs

```diff
@@ -108,6 +116,45 @@ public UIElement(string htmlTag = "div", bool isSvg = false)
 			UpdateHitTest();
 		}
 
+		private static Dictionary<Type, string> _htmlTagCache = new Dictionary<Type, string>(FastTypeComparer.Default);
+		private static Type _htmlElementAttribute;
+		private static PropertyInfo _htmlTagAttributeTagGetter;
+		private static Assembly _unoUIAssembly = typeof(UIElement).Assembly;
+
+		private string GetHtmlTag(string htmlTag)
+		{
+			var currentType = GetType();
+
+			if (currentType.Assembly != _unoUIAssembly)
+			{
+				if (_htmlElementAttribute == null)
+				{
+					_htmlElementAttribute = Assembly.Load("Uno.UI.Runtime.Wasm").GetType("Uno.UI.Runtime.Wasm.HtmlElementAttribute", true);
+					_htmlTagAttributeTagGetter = _htmlElementAttribute.GetProperty("Tag");
+				}
+
+				if (!_htmlTagCache.TryGetValue(currentType, out var htmlTagOverride))
+				{
+					htmlTagOverride = DefaultHtmlTag;
+
+					if (currentType.GetCustomAttribute(_htmlElementAttribute) is Attribute attr)
+					{
+						_htmlTagCache[currentType] = htmlTagOverride = _htmlTagAttributeTagGetter.GetValue(attr, Array.Empty<object>()) as string;
+					}
+
+					_htmlTagCache[currentType] = htmlTagOverride;
+				}
+
+				Console.WriteLine($"HTML Tag for {currentType} = {htmlTagOverride}");
```

Comment:
Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5872302)